lib/foreman/renderer.rb

```diff
@@ -26,6 +26,18 @@ def template_logger
       @template_logger ||= Foreman::Logging.logger('templates')
     end
 
+    def host_info(*path)
+      @info ||= @host.info
+      info = @info
+      path.each { |step| info = info.fetch step }
+      info
+    end
+
+    def host_param(*path)
```

Comment:
Yes, it is basically the same. The main requirement behind this was to access both params and ENC output in unified way. Also not to call methods on `@host` directly.